packages/next/src/build/webpack/config/ignore-list.ts

```diff
@@ -1,7 +1,13 @@
+import { NEXT_PROJECT_ROOT } from '../../next-dir-paths'
+
+const nextRelativePath = NEXT_PROJECT_ROOT.replace(process.cwd() + '/', '')
+
 export function shouldIgnorePath(modulePath: string): boolean {
   return (
     modulePath.includes('node_modules') ||
     // Only relevant for when Next.js is symlinked e.g. in the Next.js monorepo
-    modulePath.includes('next/dist')
+    modulePath.includes('next/dist') ||
+    // TODO: turbopack recovers the original source path of the next package
+    (!!process.env.TURBOPACK && modulePath.startsWith(nextRelativePath))
```

Comment:
Why the Turbopack check? This is a webpack only file.

Also: this file needs to move back into the config. Needless indirection. If the Turbopack middleware uses it, inline the implementation instead into the Turbopack middleware. No reason to share it at this moment since bundlers may not have the same module file names.